### PR TITLE
PRLC100-210 Need Court Permission

### PR DESCRIPTION
**Pull request by ShakiraKP**
[### JIRA link (if applicable) ###](https://tools.hmcts.net/jira/browse/PRLC100-210)

C100 Rebuild screening questions - Need permission from court to make the application

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/main/steps/c100-rebuild/permission/content.ts [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+  courtPermissionBullet2: 'guardian',
+  courtPermissionBullet3: 'special guardian',
+  courtPermissionLine3: 'You may, or may not need permission from the court.',
+  courtPermissionLine4:
+    'For example, you may need permission if you are in the children’s lives, but not their parent or guardian.',
+  courtPermissionLine5:
+    'You will need the court’s permission if you are one of the children, and the application is about you.',
+  courtPermissionLine6:
+    'If you are not sure if you need permission, <a  target="_blank" href="https://www.gov.uk/government/publications/family-court-applications-that-involve-children-cb1">see Section C of the Form CB1 guidance on GOV.UK</a>',
+  summaryText: 'Contacts for help',
```
Comment (channa-patil) [inline src/main/steps/c100-rebuild/permission/content.ts:21]: @ShakiraKP Are you using this field 'summaryText' ?
Can you please if not you are not using it. Please avoid such unused lines in the code.

File: src/main/steps/urls.ts [hunk 2]
```diff
@@ -281,6 +281,7 @@ export const C100_REASONABLE_ADJUSTMENTS_DOCUMENT_INFORMATION: PageLink = `${C10
 export const C100_REASONABLE_ADJUSTMENTS_COMMUNICATION_HELP: PageLink = `${C100_URL}/reasonable-adjustments/disability-requirements/communication-help`;
 export const C100_REASONABLE_ADJUSTMENTS_FEEL_COMFORTABLE: PageLink = `${C100_URL}/reasonable-adjustments/disability-requirements/feel-comfortable`;
 export const C100_REASONABLE_ADJUSTMENTS_TRAVELLING_COURT: PageLink = `${C100_URL}/reasonable-adjustments/disability-requirements/travelling-court`;
+export const C100_COURT_PERMISSION: PageLink = `${C100_URL}/permission`;
```
Comment (channa-patil) [inline src/main/steps/urls.ts:284]: @ShakiraKP this is part of screen questions, please append /screen-questions as base then /permission.
c100-rebuild/screening-questions/permission

File: src/main/steps/c100-rebuild/permission/template.njk [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+    <h1 class="govuk-heading-xl">{{title}}</h1>
+    <p class="govuk-body">{{courtPermissionLine1}}</p>
+    <p class="govuk-body">{{courtPermissionLine2}}</p>
+    <ul class="govuk-list govuk-list--bullet govuk-!-margin-left-7 govuk-!-margin-bottom-6">
+        <li>{{courtPermissionBullet1}}</li>
+        <li>{{courtPermissionBullet2}}</li>
+        <li>{{courtPermissionBullet3}}</li>
+    </ul>
+    <p class="govuk-body">{{courtPermissionLine3}}</p>
```
Comment (channa-patil) [inline src/main/steps/c100-rebuild/permission/template.njk:13]: @ShakiraKP Can we use an Array for these labels in content.ts & a for lop in template.njk to render them ?

File: src/main/steps/c100-rebuild/permission/template.njk [hunk 4]
```diff
@@ -0,0 +1,19 @@
+{% extends "common/template.njk" %}
+
+{% block page_content %}
+
+    <h1 class="govuk-heading-xl">{{title}}</h1>
+    <p class="govuk-body">{{courtPermissionLine1}}</p>
+    <p class="govuk-body">{{courtPermissionLine2}}</p>
+    <ul class="govuk-list govuk-list--bullet govuk-!-margin-left-7 govuk-!-margin-bottom-6">
+        <li>{{courtPermissionBullet1}}</li>
```
Comment (channa-patil) [inline src/main/steps/c100-rebuild/permission/template.njk:9]: Same as below comment. Please use an array for these common bullet points & use a for loop to render them.

File: src/main/steps/c100-rebuild/permission/content.ts [hunk 5]
```diff
@@ -0,0 +1,94 @@
+import { YesOrNo } from '../../../app/case/definition';
+
+import { TranslationFn } from './../../../app/controller/GetController';
+import { FormContent } from './../../../app/form/Form';
+import { isFieldFilledIn } from './../../../app/form/validation';
+
+const en = {
```
Comment (channa-patil) [inline src/main/steps/c100-rebuild/permission/content.ts:7]: @ShakiraKP Can you please define it as function for en & cy constants ?
For example
`const en = () => ({
  title: 'Will you be using a legal representative in these proceedings?',
});`

**Review by channa-patil — CHANGES_REQUESTED** [review]
@ShakiraKP This screen is part of screening questions, so please move all changes inside 
c100-rebuild/screening-questions/

File: src/main/steps/urls.ts (hunk 2 above)
Comment (ShakiraKP) [reply src/main/steps/urls.ts:284]: Okay no problem will adjust, I didnt know where it belonged as the link in the prototype is very basic but I will change now

File: src/main/steps/c100-rebuild/permission/template.njk (hunk 3 above)
Comment (ShakiraKP) [reply src/main/steps/c100-rebuild/permission/template.njk:13]: Okay I will look into this, just for the bullet points or for all labels?

File: src/main/steps/c100-rebuild/permission/content.ts [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+};
+
+const languages = {
+  en,
+  cy,
+};
+
+export const form: FormContent = {
+  fields: {
+    courtPermissionRequired: {
```
[the remark this answers is not in the document]
Comment (ShakiraKP) [reply src/main/steps/c100-rebuild/permission/content.ts]: @channa-patil the pages requested dont exist yet but I have added the logic

File: src/main/steps/c100-rebuild/permission/template.njk (hunk 3 above)
Comment (ShakiraKP) [reply src/main/steps/c100-rebuild/permission/template.njk]: completed

File: src/main/steps/c100-rebuild/permission/template.njk (hunk 4 above)
Comment (ShakiraKP) [reply src/main/steps/c100-rebuild/permission/template.njk]: completed

File: src/main/steps/c100-rebuild/permission/content.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (ShakiraKP) [reply src/main/steps/c100-rebuild/permission/content.ts]: completed

File: src/main/steps/c100-rebuild/screening-questions/permission/content.test.ts [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+  errors: {
+    sq_courtPermissionRequired: {
+      required:
+        'Select yes if there is any reason why you would need permission from the court to make this application',
+    },
+  },
+};
+
+const cy = {
+  title: 'Is there any reason why you would need permission from the court to make this application?',
```
Comment (yogendraU) [inline src/main/steps/c100-rebuild/screening-questions/permission/content.test.ts:32]: Can you please add - welsh at the end of each attribute .  
For e.g. Is there any reason why you would need permission from the court to make this application? - welsh

**Review by yogendraU — CHANGES_REQUESTED** [review]

File: src/main/steps/urls.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (ShakiraKP) [reply src/main/steps/urls.ts]: Completed however the pages suggested in the ticket dont actually exist yet

**Review by channa-patil — APPROVED** [review]

File: src/main/steps/c100-rebuild/screening-questions/permission/content.ts [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+
+const cy = () => ({
+  title: 'Is there any reason that you would need permission from the court to make this application? - welsh',
+  paragraphsOne: [
+    'In some cases, the court will need to give permission before the application can be made. - welsh',
+    'You will not need permission if you are the: - welsh',
+  ],
+  bulletPoints: ['parent - welsh', 'guardian - welsh', 'special guardian - welsh'],
+  paragraphsTwo: [
+    'If you are not sure if you need permission, <a  target="_blank" href="https://www.gov.uk/government/publications/family-court-applications-that-involve-children-cb1">see Section C of the Form CB1 guidance on GOV.UK</a> - welsh',
```
Comment (yogendraU) [inline src/main/steps/c100-rebuild/screening-questions/permission/content.ts:34]: @ShakiraKP , as per Jason T's comment on jira, the link is starting from Section C as below, where link stands for anchor tag:
see <link>Section C of the Form CB1 guidance</link>

**Review by yogendraU — CHANGES_REQUESTED** [review]

File: src/main/steps/c100-rebuild/screening-questions/permission/content.ts (hunk 8 above)
Comment (channa-patil) [reply src/main/steps/c100-rebuild/screening-questions/permission/content.ts]: Done

File: src/main/steps/c100-rebuild/screening-questions/permission/content.test.ts (hunk 7 above)
Comment (ShakiraKP) [reply src/main/steps/c100-rebuild/screening-questions/permission/content.test.ts]: done

**Pull request merged**